openassessment/tests/test_data.py

```diff
@@ -474,3 +478,53 @@ def test_collect_ora2_data(self):
             FEEDBACK_OPTIONS['options'][0] + '\n' + FEEDBACK_OPTIONS['options'][1]+'\n',
             FEEDBACK_TEXT,
         ])
+
+    def test_collect_ora2_responses(self):
+        self._create_submission(dict(
+            student_id=STUDENT_ID,
+            course_id=COURSE_ID,
+            item_id=ITEM_ID2,
+            item_type="openassessment"
+        ), ['self'])
+        self._create_submission(dict(
+            student_id=STUDENT_ID2,
+            course_id=COURSE_ID,
+            item_id=ITEM_ID2,
+            item_type="openassessment"
+        ), STEPS)
+
+        self._create_submission(dict(
+            student_id=STUDENT_ID,
+            course_id=COURSE_ID,
+            item_id=ITEM_ID3,
+            item_type="openassessment"
+        ), ['self'])
+        self._create_submission(dict(
+            student_id=STUDENT_ID2,
+            course_id=COURSE_ID,
+            item_id=ITEM_ID3,
+            item_type="openassessment"
+        ), ['self'])
+        self._create_submission(dict(
+            student_id=STUDENT_ID3,
+            course_id=COURSE_ID,
+            item_id=ITEM_ID3,
+            item_type="openassessment"
+        ), STEPS)
+
+        data = OraAggregateData.collect_ora2_responses(COURSE_ID)
+
+        self.assertIn(ITEM_ID, data)
+        self.assertIn(ITEM_ID2, data)
+        self.assertIn(ITEM_ID3, data)
+        for item in [ITEM_ID, ITEM_ID2, ITEM_ID3]:
+            self.assertEqual({'total', 'training', 'peer', 'self', 'staff', 'waiting', 'done'}, set(data[item].keys()))
+        self.assertEqual(data[ITEM_ID], {
+            'total': 2, 'training': 0, 'peer': 2, 'self': 0, 'staff': 0, 'waiting': 0, 'done': 0
+        })
+        self.assertEqual(data[ITEM_ID2], {
+            'total': 2, 'training': 0, 'peer': 1, 'self': 1, 'staff': 0, 'waiting': 0, 'done': 0
+        })
+        self.assertEqual(data[ITEM_ID3], {
+            'total': 3, 'training': 0, 'peer': 1, 'self': 2, 'staff': 0, 'waiting': 0, 'done': 0
+        })
```

Comment:
The items without submissions have no records in the `workflow_assessmentworkflow` table and as consequently are not returned in the results of the new `OraAggregateData.collect_ora2_responses` method.

So I don't know how to add such test because all tests here add values directly to the `workflow_assessmentworkflow`. So i think this case may stay without cover.